server/src/com/thoughtworks/go/server/controller/JobController.java

```diff
@@ -105,15 +109,22 @@ public ModelAndView jobDetail(@RequestParam("pipelineName") String pipelineName,
         StageIdentifier stageIdentifier = restfulService.translateStageCounter(pipeline.getIdentifier(), stageName, stageCounter);
 
         JobInstance instance = jobDetailService.findMostRecentBuild(new JobIdentifier(stageIdentifier, jobName));
-        return getModelAndView(instance);
+        Agent agent =  agentService.findAgentObjectByUuid(instance.getAgentUuid());
+
+        if (null == agent) {
+            agent = Agent.fromConfig(goConfigService.agentByUuid(instance.getAgentUuid()));
+        }
+
+        return getModelAndView(instance, agent);
     }
 
-    private ModelAndView getModelAndView(JobInstance jobDetail) throws Exception {
+    private ModelAndView getModelAndView(JobInstance jobDetail, Agent agent) throws Exception {
```

Comment:
The `agent` being passed is not used in the method. `presenter(jobDetail)` continues to use the one from config.